collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/MyanmarDatePickerDialog.java

```diff
@@ -63,15 +65,17 @@ protected LocalDateTime getOriginalDate() {
     private void setUpDatePicker() {
         MyanmarDate myanmarDate = MyanmarDateUtils.gregorianDateToMyanmarDate(getDateWithSkippedDaylightSavingGapIfExists());
 
-        setUpDayPicker(MyanmarDateUtils.getFirstDayOfCurrentMonth(myanmarDate), myanmarDate.getMonthDay(), myanmarDate.getMonthLength());
+        setUpDayPicker(MyanmarDateUtils.getFirstMonthDay(myanmarDate), myanmarDate.getMonthDay(), MyanmarDateUtils.getMonthLength(myanmarDate));
         setUpMonthPicker(MyanmarDateUtils.getMonthId(myanmarDate) + 1, MyanmarDateUtils.getMyanmarMonthsArray(myanmarDate.getYearInt()));
         setUpYearPicker(myanmarDate.getYearInt(), MIN_SUPPORTED_YEAR, MAX_SUPPORTED_YEAR);
     }
 
     private MyanmarDate getCurrentMyanmarDate() {
-        return MyanmarDateUtils.createMyanmarDate(
-                getYear(),
-                MyanmarDateKernel.getMyanmarMonth(getYear(), 1).getIndex().get(getMonthId()),
-                getDay());
+        List<Integer> monthIndexes = MyanmarDateKernel.getMyanmarMonth(getYear(), 1).getIndex();
+        int monthIndex = getMonthId() < monthIndexes.size() ? monthIndexes.get(getMonthId()) : monthIndexes.get(monthIndexes.size() - 1);
+        int monthLength = MyanmarDateUtils.getMonthLength(getYear(), monthIndex);
+        int dayOfMonth = getDay() > monthLength ? monthLength : getDay();
```

Comment:
This logic seems strange to me, if the day or month is too large it falls back to the max value... so every month number greater than the max is treated exactly the same?

If this is truly expected, what do you think about putting this logic inside `getDay` and `getMonthId` instead of here?

Reply:
Do we really need the ternary operator here
```
int monthIndex = getMonthId() < monthIndexes.size() ? monthIndexes.get(getMonthId()) : monthIndexes.get(monthIndexes.size() - 1);
```
When the monthPicker is set using `setUpMonthPicker` of `CustomDatePicker`, the max value is set to `monthsArray.length - 1`. So, the above check would always be true

Reply:
Both, I mean updating months and days (in that way) are needed in this case. We do the same with days in other calendars (you can check it) because for example, you can have your spinners where the selected day is 31 then you change month selecting one that contains only 30 days and we need to update days.
The same with months in Myanmar calendar, here this calendar is exceptional because different years might contain different amount of months what doesn't happen in other calendars.